File: bbos/kernel/process/thread.c
/*
 * The BBOS thread is an independent context within a process. All the threads
 * are determined at compile time and can be start and stopped at runtime, but
 * cannot be created or destroyed.
 *
 * Basically we say that thread is subprocess created by a parent process.
 *
 * Copyright (c) ???? Slade Maurer, Alexander Sviridenko
 */

#include <bbos.h>

/*
 * At least one thread should always exist, which means BBOS_NUMBER_OF_THREADS 
 * always > 0. Just a test.
 */
#if BBOS_NUMBER_OF_THREADS > 0

struct bbos_thread bbos_thread_table[BBOS_NUMBER_OF_THREADS];

/**
 * bbos_thread_init - Initialize thread.
 */
bbos_return_t
bbos_thread_init(bbos_thread_id_t id, void *action)
{
  bbos_thread_table[id].id = id;
  bbos_thread_table[id].action = action;
}

#endif /* BBOS_NUMBER_OF_THREADS > 0 */

File: 0/bbos/kernel/thread.py

__copyright__ = "Copyright (c) 2011 Slade Maurer, Alexander Sviridenko"

from bbos.component import Component

#_______________________________________________________________________________

class Thread(Component):
    """Thread base class."""
    def __init__(self, name, entry, alias=None):
        Component.__init__(self, name)
        self.entry = entry
        if not alias:
            alias = entry
        self.set_alias(alias)
    # __init__()

    def set_alias(self, alias):
        self.alias = alias
    # set_alias()

    def get_entry(self):
        """Get thread entry function."""
        return self.entry
    # get_entry()

    def get_alias(self):
        """Get alias of entry function."""
        return self.alias
    # get_alias()

